### DISCUSSION TITLE
fief-python types

### DISCUSSION THREAD

**Discussion by davidbrochart** [fief-dev/fief#168] [Q&A] [answered]
I know `fief-python` has a [py.typed](https://github.com/fief-dev/fief-python/blob/1773c7e2fa6dfd931f33cc718ec1aaa80dbe58be/fief_client/py.typed), but `mypy` (v1.1.1) fails on:
```py
import fief_client
```
with:
```
foo.py:1: error: Cannot find implementation or library stub for module named "fief_client"  [import]
foo.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```
Any idea why?

**Comment by frankie567** [comment]
I was able to reproduce it, that's indeed really strange and annoying 😟 I'll try to see if there is something missing somewhere.

**Comment by frankie567** [comment, marked as answer]
Lol, ok, this is nasty 😂

It's probably because you have installed `mypy` **outside** a virtual environment. Even if you install it in a virtual environment, when you invoke `mypy`, it'll still call the one from the global environment which doesn't have `fief_client`.

If you do `pip uninstall mypy` while outside your virtual environment and try again, it should work properly.

It's fun because we fell exactly into the same trap 😅

**Reply by davidbrochart** [reply]
Indeed, thanks!
